Apps/WebClient/src/ClientApp/src/components/modal/rating.vue

```diff
@@ -0,0 +1,105 @@
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.config";
+
+@Component
+export default class RatingComponent extends Vue {
+    private logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+    @Prop({
+        default:
+            "Did the Health Gateway improve your access to health information today? Please provide a rating.",
+        required: false,
+    })
+    question!: string;
+    private ratingValue: number = 0;
+    private isVisible: boolean = false;
+    public showModal() {
+        this.isVisible = true;
+    }
+    public hideModal() {
+        this.isVisible = false;
+    }
+    private handleRating(value: number, skip: boolean = false) {
+        this.logger.debug(
+            `Emitting the selected rating value: ${value} to the modal's caller ...`
+        );
+        this.$emit("selected-rating", { ratingValue: value, skip: skip });
+        this.hideModal();
+    }
+    private skip() {
+        this.logger.debug(`clicked on skip button (value: 0)...`);
+        this.handleRating(0, true);
+    }
+}
+</script>
+
+<template>
+    <div>
+        <b-modal
+            id="rating-modal"
+            ref="rating-modal"
+            v-model="isVisible"
+            title="Rating"
+            size="md"
+            header-bg-variant="primary"
+            header-text-variant="light"
+            footer-class="modal-footer"
+            no-close-on-backdrop
+            hide-header-close
+            no-close-on-esc
+            centered
+        >
+            <b-row class="text-center">
+                <b-col>
+                    {{ question }}
+                </b-col>
+            </b-row>
+            <b-row class="text-center row-answer">
+                <b-col>
+                    <b-form-rating
+                        v-model="ratingValue"
+                        variant="warning"
+                        class="mb-2"
+                        no-border
+                        size="lg"
+                        @change="handleRating(ratingValue)"
+                    ></b-form-rating>
+                </b-col>
+            </b-row>
+            <template v-slot:modal-footer>
+                <b-row>
+                    <b-col>
+                        <b-row>
+                            <b-col>
+                                <b-button
+                                    id="skipButton"
+                                    variant="outline-primary"
+                                    @click="skip()"
+                                    >Skip</b-button
+                                >
+                            </b-col>
+                        </b-row>
+                    </b-col>
+                </b-row>
+            </template>
+        </b-modal>
+    </div>
+</template>
+
+<style lang="scss" scoped>
+@import "@/assets/scss/_variables.scss";
+.modal-content {
+    max-width: 500px;
+    .modal-body {
```

Comment:
did this work at all?

Reply:
this works well
![image](https://user-images.githubusercontent.com/48332333/90549113-de2fff00-e142-11ea-9303-3fd4791de3d7.png)